### DISCUSSION TITLE
Migrating an existing site to Jekyll

### DISCUSSION THREAD

**Discussion by Gineus** [jekyll/jekyll#9598] [Q&A]
Hi there,

I would like to migrate an existing website to Jekyll. The current website consists of a database containing articles. The articles have one assigned category. Each article has its own URL (in order for users to bookmark the article). There are also views by category, where the headline of the article is displayed as link and points to the page of the respective article. The category view is paged so that not all articles for a category must be loaded (performance) but also that the user is not overwhelmed with thousands of articles at once (usability). In the web page you can also rate articles (rating is saved in the DB), but I guess I cannot achieve this behaviour in a static webpage so it's fine to not address this if not possible.

My questions are:
* Are the things I want to achieve possible with Jeykll?
* What ways/steps do you recommend to migrate the existing website?

Thanks in advance.

**Comment by guillermodotn** [comment]
Hello @Gineus,

Yes, as you mentioned, **Jekyll** can provide all the features you’ve listed above. Regarding the rating system, you might want to check out [Giscus](https://github.com/giscus/giscus). While it's not exactly a rating system, it allows for comments and reactions on your blog, which could serve a similar purpose.

As for the migration steps, I’d recommend starting by selecting a theme you like (or creating your own). Then, create a script to extract your blog post data from the database and generate the corresponding post files in **Jekyll**.
